mlserver/logging.py

```diff
@@ -17,7 +19,12 @@ def get_logger():
 
 
 def apply_logging_file(logging_settings: str):
-    logging.config.fileConfig(fname=logging_settings, disable_existing_loggers=False)
+    if "json" in Path(logging_settings).suffix:
```

Comment:
Should we add a third case where `logging_settings` contains the dictionary directly (as opposed to the file name with the dictionary)?

Reply:
Thanks for taking a look at this @adriangonz! I've added some logic to support all 3 options...what do you think?

It's a bit messier now, with `logging_settings` potentially being either a `str` or `Dict`, but I like the added functionality. Let me know if you have any suggestions for how this might be cleaned up / if that's needed!

```python
def apply_logging_file(logging_settings: Union[str, Dict]):
    if isinstance(logging_settings, str) and Path(logging_settings).is_file():
        if "json" in Path(logging_settings).suffix:
            with open(logging_settings) as settings_file:
                config = json.load(settings_file)
            logging.config.dictConfig(config)
        else:
            logging.config.fileConfig(fname=logging_settings, disable_existing_loggers=False)
    elif isinstance(logging_settings, Dict):
        logging.config.dictConfig(logging_settings)
    else:
        logger.warning("Unable to parse logging_settings.")
```

Reply:
Nice one @dumaas ! I didn't realise that supporting the 3rd case would also mean supporting multiple types there. I think you worked your way around that beautifully though! Changes look great! :rocket: 

Once the tests are green, this should be good to go :+1:

Reply:
Thank you! 😄 

@adriangonz I believe this commit should resolve the linting errors: https://github.com/SeldonIO/MLServer/pull/742/commits/6ea65673512ac175e0d2cf5e43d067fd0a0140b5